### ISSUE TITLE
Testing custom work dependencies example fails due to timeout in CI

### ISSUE THREAD

**Issue by akihironitta** [Lightning-AI/lightning#15417]
### Bug description

One of our app example tests fails due to timeout.

- CI file: https://github.com/Lightning-AI/lightning/blob/c39c8eb2e4f49e35ef38a2ae6de2765ec71623db/.azure/app-cloud-e2e.yml#L66
- CI run: https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=111050&view=logs&j=4e1b52d3-134d-51bf-b10a-6dde7facebae&t=60ace6bb-5e20-5573-9db7-5550ec53d684&l=1463

### Error messages and logs

```
...
flow 10/30/2022 17:05:43 DEBUG: The url http://10.1.113.254:47157/ is not ready. 
flow 10/30/2022 17:05:43 DEBUG: The url http://10.1.113.254:47157/ is not ready. 
flow 10/30/2022 17:05:43 DEBUG: The url http://10.1.113.254:47157/ is not ready. 
(node:446) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 end listeners added to [ReadStream]. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
(node:446) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added to [ReadStream]. Use emitter.setMaxListeners() to increase limit
/bin/sh: 1: ***: not found
FAILED

=================================== FAILURES ===================================
_________________ test_custom_work_dependencies_example_cloud __________________

    @pytest.mark.cloud
    def test_custom_work_dependencies_example_cloud() -> None:
        # if requirements not installed, the app will fail
        with run_app_in_cloud(
            os.path.join(_PROJECT_ROOT, "tests/tests_app_examples/custom_work_dependencies/"),
            app_name="app.py",
        ) as (_, _, fetch_logs, _):
            has_logs = False
            while not has_logs:
>               for log in fetch_logs(["flow"]):

tests/tests_app_examples/test_custom_work_dependencies.py:19: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/***/app/testing/testing.py:440: in fetch_logs
    for log_event in gen:
src/***/app/utilities/app_logs.py:96: in _app_logs_reader
    log_event: _LogEvent = read_queue.get(timeout=None if follow else 1.0)
/usr/lib/python3.8/queue.py:179: in get
    self.not_empty.wait(remaining)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Condition(<unlocked _thread.lock object at 0x7fe4bb27f300>, 0)>
timeout = 0.9999995000000581

    def wait(self, timeout=None):
        """Wait until notified or until a timeout occurs.
    
        If the calling thread has not acquired the lock when this method is
        called, a RuntimeError is raised.
    
        This method releases the underlying lock, and then blocks until it is
        awakened by a notify() or notify_all() call for the same condition
        variable in another thread, or until the optional timeout occurs. Once
        awakened or timed out, it re-acquires the lock and returns.
    
        When the timeout argument is present and not None, it should be a
        floating point number specifying a timeout for the operation in seconds
        (or fractions thereof).
    
        When the underlying lock is an RLock, it is not released using its
        release() method, since this may not actually unlock the lock when it
        was acquired multiple times recursively. Instead, an internal interface
        of the RLock class is used, which really unlocks it even when it has
        been recursively acquired several times. Another internal interface is
        then used to restore the recursion level when the lock is reacquired.
    
        """
        if not self._is_owned():
            raise RuntimeError("cannot wait on un-acquired lock")
        waiter = _allocate_lock()
        waiter.acquire()
        self._waiters.append(waiter)
        saved_state = self._release_save()
        gotit = False
        try:    # restore state no matter what (e.g., KeyboardInterrupt)
            if timeout is None:
                waiter.acquire()
                gotit = True
            else:
                if timeout > 0:
>                   gotit = waiter.acquire(True, timeout)
E                   Failed: Timeout >1200.0s

/usr/lib/python3.8/threading.py:306: Failed
```

```
=========================== short test summary info ============================
FAILED tests/tests_app_examples/test_custom_work_dependencies.py::test_custom_work_dependencies_example_cloud
======================== 1 failed in 1373.06s (0:22:53) ========================

+++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++

~~~~~~~~~~~~~~~~~~~~ Stack of Thread-241 (140620377880320) ~~~~~~~~~~~~~~~~~~~~~
  File "/usr/lib/python3.8/threading.py", line 890, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/home/AzDevOps_azpcontainer/.local/lib/python3.8/site-packages/websocket/_app.py", line 449, in run_forever
    setSock()
  File "/home/AzDevOps_azpcontainer/.local/lib/python3.8/site-packages/websocket/_app.py", line 378, in setSock
    dispatcher.read(self.sock.sock, read, check)
  File "/home/AzDevOps_azpcontainer/.local/lib/python3.8/site-packages/websocket/_app.py", line 72, in read
    r = sel.select(self.ping_timeout)
  File "/usr/lib/python3.8/selectors.py", line 468, in select
    fd_event_list = self._selector.poll(timeout, max_ev)

~~~~~~~~~~~~~~~~~~~~~ Stack of waitpid-0 (140621050640128) ~~~~~~~~~~~~~~~~~~~~~
  File "/usr/lib/python3.8/threading.py", line 890, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.8/asyncio/unix_events.py", line 1318, in _do_waitpid
    pid, status = os.waitpid(expected_pid, 0)

+++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++
##[error]Bash exited with code '1'.
```

cc @tchaton @rohitgr7 @carmocca @akihironitta @borda

**Comment by akihironitta** [comment]
Marked this issue as `priority: 0` as it blocks all ongoing PRs that trigger this test. Could someone please have a look?

**Comment by akihironitta** [comment]
Closing this issue as per #15431. https://dev.azure.com/Lightning-AI/lightning/_build/results?buildId=111417&view=logs&j=4e1b52d3-134d-51bf-b10a-6dde7facebae

**Comment by akihironitta** [comment]
> I think that https://github.com/Lightning-AI/lightning/pull/15431 was a hotfix to unblock the master but not addressing the underlying issue...

Reopening this issue as commented #15429. May be fixed by #15429.

**Comment by akihironitta** [comment]
Changed the label to P1 as this no longer blocks master.

**Comment by akihironitta** [comment]
Closing this as it's already been fixed by #15431. Thank you @manskx for your help :)
